src/type_checker/mod.rs

```diff
@@ -208,7 +207,7 @@ impl<B: Backend> TypeChecker<B> {
                             ConstInfo {
                                 value: vec![cst.value],
                                 typ: Ty {
-                                    kind: TyKind::Field,
+                                    kind: TyKind::Field { constant: true },
```

Comment:
Some thoughts: in theory we can have constants of any type. But now we only support `Field` constants. 

Also, a non-`Field` value can easily be checked to be a constant or not by checking if all of its `Field` fields are constants (because down the line a type is constituted of many `Field`s). So it should be straight forward to support non-`Field` consts at some point.

I don't think it's urgent though, so I will not create an issue for that